client/my-sites/email/email-providers-comparison/price/google-workspace.tsx

```diff
@@ -62,39 +83,62 @@ const GoogleWorkspacePrice = ( {
 	}
 
 	const isDiscounted = hasDiscount( product );
-	const isEligibleForFreeTrial = isDomainEligibleForGoogleWorkspaceFreeTrial( { domain, product } );
 
 	const priceWithInterval = (
 		<PriceWithInterval
 			currencyCode={ currencyCode ?? '' }
 			intervalLength={ intervalLength }
 			isDiscounted={ isDiscounted }
-			isEligibleForFreeTrial={ isEligibleForFreeTrial }
+			isEligibleForIntroductoryOffer={ isEligibleForIntroductoryOffer }
 			product={ product }
 		/>
 	);
 
 	return (
 		<>
-			{ isDiscounted && (
-				<div className="google-workspace-price__discount-badge badge badge--info-green">
-					{ translate( 'Limited time: %(discount)d%% off', {
+			{ isEligibleForIntroductoryOfferFreeTrial && (
+				<div className="google-workspace-price__trial-badge badge badge--info-green">
+					{ translate( '1 month free' ) }
+				</div>
+			) }
+
+			{ ! isEligibleForIntroductoryOfferFreeTrial && isEligibleForIntroductoryOffer && (
+				<div className="google-workspace-price__trial-badge badge badge--info-green">
+					{ translate( '%(approximateDiscountForOffer)d%% off', {
 						args: {
-							discount: product?.sale_coupon?.discount,
+							approximateDiscountForOffer: getApproximateDiscountForOffer( product ),
 						},
-						comment: "%(discount)d is a numeric discount percentage (e.g. '40')",
+						comment: "%(approximateDiscountForOffer)d is a numeric discount percentage (e.g. '40')",
 					} ) }
 				</div>
 			) }
 
-			{ isEligibleForFreeTrial && (
-				<div className="google-workspace-price__trial-badge badge badge--info-green">
-					{ translate( '1 month free' ) }
+			{ isDiscounted && ! isEligibleForIntroductoryOfferFreeTrial && (
```

Comment:
Let's place the `! isEligibleForIntroductoryOfferFreeTrial` check first so it's consistent with the other conditions :)